### PR TITLE
 complete task 6

### PR DESCRIPTION
**Pull request by Tanim1995**
created a controller and model for comments
created 3 separate tests for verify if I could get the comments based on id and test invalid id and not an id

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -147,6 +143,39 @@ describe("GET/api/articles/:article_id", () => {
...
+        expect(body.comments).toBeSortedBy("created_at");
+        body.comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            title: expect.any(String),
+            article_id: expect.any(Number),
+            created_at: expect.any(String),
+            votes: expect.any(Number),
+            body: expect.any(String),
+          });
```
Comment (MKD01) [inline __tests__/app.test.js:161]: This is a really good start but dont forget we also want to ensure the array is not empty so before the forEach you will need to assert that the array has atleast one element within it. We also should be expecting the article_id for all of these to be 3 as that those are the comments we are expecting back.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -147,6 +143,39 @@ describe("GET/api/articles/:article_id", () => {
...
+      });
+  });
+  test("response with the status of 400 and an appropriate error message if parameter id is not an id ", () => {
+    return request(app)
+      .get("/api/articles/notAnId/comments")
+      .expect(400)
+      .then((res) => {
+        expect(res.body.message).toBe("Bad Request");
+      });
+  });
```
Comment (MKD01) [inline __tests__/app.test.js:180]: Really good error handling.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -147,6 +143,39 @@ describe("GET/api/articles/:article_id", () => {
...
 });
-
```
Comment (MKD01) [inline __tests__/app.test.js:151]: One very important test which is missing here is a test to ensure when given a valid article id but one with no comments associated with it we receive an empty array. The endpoint for this will be /api/articles/2/comments

**Review by MKD01 — COMMENTED** [review]
Really good work but there are some changes/a missing test that should be added for this test. Please do make those requested changes and merge them in without needing to send an updated pr for them.
